# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmStubs.cpp
# Commit: 9a162cc369e3
# Full Hash: 9a162cc369e3b6f51f5767924e63d46539c4f631
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-02-22 09:14:45
# Regressor Bug: 1861078
# File Overlap Count: 1
# Description:
#   Bug 1861078 - wasm: make metadata offsets relative to CodeBlock->codeBase. r=rhunt
#   
#   Metadata used offsets to the beginning of code segment. Change that to use
#   offsets relative to the beginning of CodeBlock. This allows us to avoid offseting
#   metadata later on and prepare the work for simplification of stackmaps.
# ==============================================================================

diff -r e543a77786c9 -r 9a162cc369e3 js/src/wasm/WasmStubs.cpp
--- a/js/src/wasm/WasmStubs.cpp	Fri Feb 21 17:09:51 2025 +0000
+++ b/js/src/wasm/WasmStubs.cpp	Fri Feb 21 17:17:58 2025 +0000
@@ -1516,7 +1516,8 @@
   // Actual call.
   const CodeBlock& codeBlock = inst.code().funcCodeBlock(fe.funcIndex());
   const CodeRange& codeRange = codeBlock.codeRange(fe);
-  void* callee = codeBlock.segment->base() + codeRange.funcUncheckedCallEntry();
+  void* callee = const_cast<uint8_t*>(codeBlock.base()) +
+                 codeRange.funcUncheckedCallEntry();
 
   masm.assertStackAlignment(WasmStackAlignment);
   MoveSPForJitABI(masm);
